packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -194,6 +199,11 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 	}
 
 	private submitCommit(commit: Commit<TChange>): void {
+		// Nested transactions are tracked as part of the outermost transaction
+		if (this.transactions.size === 0) {
```

Comment:
```suggestion
		if (!this.isTransacting()) {
```

Reply:
But also, I think there is a bug right now that I wrote. `submitCommit` should only be called when the root transaction commits, but it looks like we call it for all nested transactions? Yikes. Once I fix that, you should be able to make this conditional an assert instead.